### DISCUSSION TITLE
qutebrowser v3.4.0 released / 11 years qutebrowser!

### DISCUSSION THREAD

**Discussion by The-Compiler** [qutebrowser/qutebrowser#8423] [General]
I'm delighted that qutebrowser is 11 years old today, almost on the
minute:

    Author: Florian Bruhin <git@the-compiler.org>
    Date:   Sat Dec 14 22:15:16 2013 +0100

        Initial commit

If you're feeling nostalgic, in 2022 I did a little writeup about how it
all started:
https://listi.jpberlin.de/pipermail/qutebrowser-announce/2022-December/000115.html

What better way to celebrate than with a new release? So I just released
v3.4.0 (the CI had other plans, but on the 5th try it finally worked).

The main highlight in this release is probably proper Qt 6.8 support
finally, including asking the user for clipboard permission on-demand
instead of needing to grant that before clipboard buttons start working.

There also were a couple of bugfixes (one of them improving website
compatibility when they do XHR requests with a custom Accept-Language
header), and Windows/macOS releases finally ship with Qt 6.8 (PyQt 6.8
was a bit delayed and only released two days ago).

Nothing else too big in there, but I'm hoping we'll get around to some
bigger topics in 2025! toofar has been looking at getting tree-style
tabs finished finally, and personally there are a variety of topics I'd
love to have a look at. We'll see how it all pans out!

Here's the full changelog:

## Removed

- Support for Python 3.8 is dropped, and Python 3.9 is now required. (#8325)
- Support for macOS 12 Monterey is now dropped, and binaries will be built on
  macOS 13 Ventura. (#8327)
- When using the installer on Windows 10, build 1809 or newer is now required
  (previous versions required 1607 or newer, but that's not officialy supported by
  Qt upstream). (#8336)

## Changed

- Windows/macOS binaries are now built with Qt 6.8.1. (#8242)
    - Based on Chromium 122.0.6261.171
    - With security patches up to 131.0.6778.70
- Windows/macOS binaries are now using Python 3.13. (#8205)
- The `.desktop` file now also declares qutebrowser as a valid viewer for
  `image/webp`. (#8340)
- Updated mimetype information for getting a suitable extension when downloading
  a `data:` URL.
- The `content.javascript.clipboard` setting now defaults to "ask", which on
  Qt 6.8+ will prompt the user to grant clipboard access. On older Qt versions,
  this is still equivalent to `"none"` and needs to be set manually. (#8348)
- If a XHR request made via JS sets a custom `Accept-Language` header, it now
  correctly has precedence over the global `content.headers.accept_language`
  setting (but not per-domain overrides). This fixes subtle JS issues on
  websites that rely on the custom header being sent for those requests, and
  e.g. block the requests server-side otherwise. (#8370)
- Our packaging scripts now prefer the "legacy"/"for older browsers" PDF.js
  build as their normal release only supports the latest Chromium version and
  might break in qutebrowser on updates. **Note to packagers:** If there's a
  PDF.js package in your distribution as an (optional) qutebrowser dependency,
  consider also switching to this variant (same code, built differently).

## Fixed

- Crash with recent Jinja/Markupsafe versions when viewing a finished userscript
  (or potentially editor) process via `:process`.
- `scripts/open_url_in_instance.sh` now avoids `echo -n`, thus running
  correctly on POSIX sh. (#8409)
- Added a workaround for a bogus QtWebEngine warning about missing spell
  checking dictionaries. (#8330)

Enjoy!

**Comment by origi0** [comment]
Yes! "11 years nostalgic" ah moments
